### PR TITLE
Use container agent for CI build

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Use container agent for CI job

https://github.com/jenkinsci/bom/pull/1338#issuecomment-1198016897 notes that there is an automated test failure in the bill of materials and plugin compatibility tests.  This change is trying to duplicate that failure on ci.jenkins.io.

The first commit in this pull request will be considered "successful" if the build status is failing for the first run.

* [Build Status](https://ci.jenkins.io/job/Plugins/job/git-client-plugin/view/change-requests/job/PR-880/1/) (build 1 aborted for build 2)
* [Build Status](https://ci.jenkins.io/job/Plugins/job/git-client-plugin/view/change-requests/job/PR-880/2/) - build 2 (confirms problem visible on ci.jenkins.io)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/jenkinsci/git-client-plugin/blob/master/CONTRIBUTING.adoc) doc
- [x] I have referenced the GitHub comment related to my changes in one or more commit messages
- [x] Unit tests pass locally with my changes
- [x] I have added documentation as necessary
- [x] No Javadoc warnings were introduced with my changes
- [x] No spotbugs warnings were introduced with my changes

## Types of changes

- [x] Infrastructure change (non-breaking change which updates dependencies or improves infrastructure)

### CODE REVIEW

**Review by Pldi23 — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!groovy
 
 buildPlugin(failFast: false,
+            useContainerAgent: true,
```
Comment (jglick) [inline Jenkinsfile:4]:
```suggestion
```

**Review by jglick — COMMENTED** [review]
May as well run a mixture of container- and VM-based tests.

**Comment by Pldi23** [comment]
`org.jenkinsci.plugins.gitclient.verifier.NoHostKeyVerifierTest#testVerifyServerHostKey` which do not verify keys at all and simply return true failed with  `Caused by: java.net.ConnectException: Connection timed out (Connection timed out)`.

Kind of we can't connect from container using ssh...

**Comment by MarkEWaite** [comment]
@Pldi23 I've run out of time that I can spend on this investigation.  I've made multiple guesses and tried multiple alternatives without understanding the root cause.  I've tried closing the JGitConnection explicitly in the test.  I've tried removing the parallel execution with forks.  I've tried reducing dependencies.  No change.

There was a period where the tests were passing on `maven-11`, then they started failing again.  I don't know what changed.

I've added conditionals that exit the tests early if the test is running on a Kubernetes agent on ci.jenkins.io.  By limiting the exclusion to only kubernetes on ci.jenkins.io, that will allow the tests to continue running on my cluster.

I'd like to deliver git client plugin 3.12.0 from the master branch early this week so that JGit 5.13.1 is available to Jenkins users and so that the master branch represents a recent delivery.  Does that seem reasonable to you?

**Comment by Pldi23** [comment]
>I'd like to deliver git client plugin 3.12.0 from the master branch early this week so that JGit 5.13.1 is available to Jenkins users and so that the master branch represents a recent delivery. Does that seem reasonable to you?

@MarkEWaite fine for me to deliver 3.12.0. (When I have some free time, I will also try to figure out what could be the reason here)

**Pull request merged**